### ISSUE TITLE
can't reproduce the SwAV STL-10 example

### ISSUE THREAD

**Issue by chtlp** [PyTorchLightning/pytorch-lightning#7325]
## 🐛 Bug

Hi, I can't reproduce the SwAV STL-10 model results using instructions from the document:
https://pytorch-lightning-bolts.readthedocs.io/en/latest/self_supervised_models.html#id21

### To Reproduce

Because I don't have a V100, I am training with 1 or 2 2080ti. I restricted the batch size to 64 because of the smaller GPU memory.

Training with 2 GPUs:
```
python -m vault_embed.swav.swav_module \
    --online_ft \
    --gpus 2 \
    --optimizer lars \
    --batch_size 64 \
    --learning_rate 1e-3 \
    --gaussian_blur \
    --queue_length 0 \
    --jitter_strength 1. \
    --nmb_prototypes 512 \
    --dataset stl10 --data_dir /run/shm/stl-10 \
    --fast_dev_run 0
```

Training with 1 GPU:
```
python -m vault_embed.swav.swav_module \
    --online_ft \
    --gpus 1 \
    --optimizer lars \
    --batch_size 64 \
    --learning_rate 1e-3 \
    --gaussian_blur \
    --queue_length 0 \
    --jitter_strength 1. \
    --nmb_prototypes 512 \
    --dataset stl10 --data_dir /run/shm/stl-10 \
    --fast_dev_run 0
```

The training all stuck at a high training loss without making further progress

```
Epoch 44:  99% 778/783 [27:11<00:10,  2.10s/it, loss=6.24, v_num=13]
Epoch 44:  99% 779/783 [27:11<00:08,  2.09s/it, loss=6.24, v_num=13]
Epoch 44: 100% 780/783 [27:12<00:06,  2.09s/it, loss=6.24, v_num=13]
Epoch 44: 100% 781/783 [27:12<00:04,  2.09s/it, loss=6.24, v_num=13]
```

![Xnip2021-05-03_18-12-52](https://user-images.githubusercontent.com/568535/116864912-352b2500-ac3b-11eb-9a8d-1fbe1f73ab53.jpg)

Really important for my research to get this model working, let me know if more information should be provided. Many thanks!!

### Environment

```
python collect_env_details.py
```

* CUDA:
	- GPU:
		- GeForce RTX 2080 Ti
		- GeForce RTX 2080 Ti
		- GeForce RTX 2080 Ti
		- GeForce RTX 2080 Ti
		- GeForce RTX 2080 Ti
		- GeForce RTX 2080 Ti
	- available:         True
	- version:           10.2
* Packages:
	- numpy:             1.20.0
	- pyTorch_debug:     False
	- pyTorch_version:   1.8.1+cu102
	- pytorch-lightning: 1.2.10
	- tqdm:              4.60.0
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- ELF
	- processor:         x86_64
	- python:            3.9.1
	- version:           #202011181948 SMP Wed Nov 18 20:49:19 UTC 2020

**Comment by awaelchli** [comment]
Hi
I pinged @ananyahjha93, he may be able to tell us if he saw different results for training on one vs. multiple GPUs. In theory multi-gpu with the same effective batch size should give comparable results.

**Comment by ananyahjha93** [comment]
@chtlp I think the problem is related to the optimizer selection. Earlier we used Adam with a LARS wrapper to provide layer adaptation which made a learning rate of 1e-3 suitable. Now LARS has been updated to be an Optimizer class object instead of being a wrapper class around Optimizers. That sets the default optimizer to SGD and applies layer adaptation on top of that. From my experience a learning rate of 1e-3 would be highly unsuitable for SGD. 

Where you can start right now is by setting the learning rates to 0.01, 0.05 and 0.1. I'll try and see if we can run a learning rate search experiment quickly and update the docs.

**Comment by chtlp** [comment]
Thanks @awaelchli @ananyahjha93 ! 

I think the parameter on lars indeed needs to be updated in the document:
https://pytorch-lightning-bolts.readthedocs.io/en/latest/self_supervised_models.html#id21
![Xnip2021-05-03_20-54-01](https://user-images.githubusercontent.com/568535/116878295-bbeafc80-ac51-11eb-9cd0-bbc70e8607c2.jpg)


I will try some different settings, but I do have limited GPU resources at hand right now and a recommended parameter setting that can reproduce the results would be appreciated!

**Comment by chtlp** [comment]
@ananyahjha93 I tried to set the learning rate to 0.1, but still no luck yet. Can we still use ADAM with LARS?

```
PL_EXP_VERSION=swav_stl10_gpu1_lars01 CUDA_VISIBLE_DEVICES=1,2 python -m vault_embed.swav.swav_module \
    --online_ft \
    --gpus 2 \
    --optimizer lars \
    --batch_size 64 \
    --learning_rate 0.1 \
    --gaussian_blur \
    --queue_length 0 \
    --jitter_strength 1. \
    --nmb_prototypes 512 \
    --dataset stl10 --data_dir /run/shm/stl-10 \
     --fast_dev_run 0
```
![Xnip2021-05-03_23-18-41](https://user-images.githubusercontent.com/568535/116895578-ef378680-ac65-11eb-960a-aa7465113d0d.jpg)

**Comment by awaelchli** [comment]
You can still use adam yes:
`--optimizer adam`

We will rerun experiments and try to find the best learning rate for lars.

**Comment by JACKHAHA363** [comment]
Is there any update on this?
